core/table.ts

```diff
@@ -225,6 +235,7 @@ export interface ITableContext {
   disabled: () => string;
   redshift: (redshift: dataform.IRedshiftOptions) => string;
   bigquery: (bigquery: dataform.IBigQueryOptions) => string;
+  sqldatawarehouse: (sqldatawarehouse: dataform.ISQLDataWarehouseOptions) => string;
```

Comment:
i'd leave this out of `ITableContext`, these other methods are here mostly to support non-SQLX usage, as commented elsewhere i don't think we want to add to that legacy stuff. for SQLX and JS it should be enough to include it in `TConfig` and to have your new method defined on `Table`.